apps/speed.c

```diff
@@ -1364,44 +1364,40 @@ int speed_main(int argc, char **argv)
     int dsa_doit[DSA_NUM] = { 0 };
 #endif
 #ifndef OPENSSL_NO_EC
+    typedef struct {
+        const char *name;
+        unsigned int nid;
+        unsigned int bits;
+    } curve_nameid_t;
     /*
      * We only test over the following curves as they are representative, To
      * add tests over more curves, simply add the curve NID and curve name to
      * the following arrays and increase the EC_NUM value accordingly.
      */
-    static const unsigned int test_curves[EC_NUM] = {
+    static const curve_nameid_t test_curves[EC_NUM] =
```

Comment:
I wish it was possible to omit EC_NUM. But at least, while you are at it, would you declare this array as [], and insert OPENSSL_assert that compares number of elements to EC_NUM? So that it works as remainder when you add things to it...

Reply:
Oups, will looking at this more carefully, I found a larger issue.
There is :  
- 18 ecdh curves as EC_NUM states
- and only ecdsa 16 curves but it reuse EC_NUM 

Fix in progress.